app/models/govspeak_decorator.rb

```diff
@@ -0,0 +1,28 @@
+class GovspeakDecorator < DelegateClass(Govspeak::Document)
+  ALLOWED_TAGS = %w[details summary p h1 h2 h3 h4 ul li img div ol a span strong iframe].freeze
+
+  def self.translate_markdown(markdown, sanitize: true)
+    newdoc = new(Govspeak::Document.new(markdown, sanitize: sanitize, allowed_elements: ALLOWED_TAGS))
+    newdoc.to_html
+  end
+
+  Govspeak::Document.extension('YoutubeVideo', /\$YoutubeVideo(?:\[(.*?)\])?\((.*?)\)\$EndYoutubeVideo/m) do |title, youtube_link|
+    youtube_id = youtube_link.scan(/^.*(youtu.be\/|v\/|u\/\w\/|embed\/|watch\?v=|&v=)([^#&?]*).*/)[0][1]
+    embed_url = %(https://www.youtube.com/embed/#{youtube_id}?enablejsapi=1&amp;origin=https://help-for-early-years-providers.education.gov.uk)
```

Comment:
Do we need to make the `origin` configurable?

Reply:
Yes Agree and have updated this now